### ISSUE TITLE
Error While extracting Non-English tables (Arabic) reversed Text

### ISSUE THREAD

**Issue by iiAmeer** [jsvine/pdfplumber#1159]
I have a problem, when i use extract_tables function, i get a reversed arabic text. Like the actual name is joe, but i get eoj. (it's arabic but i wrote english so u can understand), there is no need for screenshots its clear.

## Environment

- pdfplumber version: latest 
- Python version: 3.11
- OS: Windows (later linux)

**Comment by jsvine** [comment]
Thanks for noting this. Could you share the PDF? That will make it easier to diagnose the issue and suggest solutions.

**Comment by iiAmeer** [comment]
> Thanks for noting this. Could you share the PDF? That will make it easier to diagnose the issue and suggest solutions.

![087_ثانوية الإمام علي بن ابي طالب الأهلية للبنات (1).pdf](https://github.com/user-attachments/files/15975408/087_.1.pdf)

This is the file i used, if you want my code & run preview wait me until tomorrow . 

![IMG_20240626_092555.jpg](https://github.com/jsvine/pdfplumber/assets/68138224/a51a51eb-f0b2-4b62-86f7-aafebffd7a1e)

**Comment by jsvine** [comment]
Thanks for providing the PDF. Ignoring the numbers for now, and instead focusing on the text ... does the output of this look more like what you want?:

```python
import pandas as pd
table = page.extract_table(dict(text_char_dir_render="rtl"))
pd.DataFrame(table, columns=None).fillna("")
```
<img width="854" alt="Screenshot 2024-07-01 at 11 09 20 PM" src="https://github.com/jsvine/pdfplumber/assets/534702/000b3004-9d3e-45a8-93b2-7e9532fdac06">

**Comment by iiAmeer** [comment]
> Thanks for providing the PDF. Ignoring the numbers for now, and instead focusing on the text ... does the output of this look more like what you want?:
> 
> ```python
> import pandas as pd
> table = page.extract_table(dict(text_char_dir_render="rtl"))
> pd.DataFrame(table, columns=None).fillna("")
> ```
> 
> <img alt="Screenshot 2024-07-01 at 11 09 20 PM" width="854" src="https://private-user-images.githubusercontent.com/534702/344862534-000b3004-9d3e-45a8-93b2-7e9532fdac06.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjExMzc1ODcsIm5iZiI6MTcyMTEzNzI4NywicGF0aCI6Ii81MzQ3MDIvMzQ0ODYyNTM0LTAwMGIzMDA0LTlkM2UtNDVhOC05M2IyLTdlOTUzMmZkYWMwNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxNlQxMzQxMjdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NTFmZDFkZThhYjVlMGMxZTI0NjEyM2Y1NzNiZTJhODE2ZTBkMGU5OGVjNGM3OTM5ODcwZTRiYzQ4YTU3Mzc2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9._yBbr0gnGMueuMmb4F4JmENsR3E3iTilbnRe-hWnEQg">

yes, exactlly. BUT numbers are IMPORTANT and can't be reversed. Note second col (13 in ur viewer) refer to index , so it can't be reversed. Please apply reverstion in RTL only langs like Arabic,
Aramaic.
Azeri.
Dhivehi/Maldivian.
Hebrew.
Kurdish (Sorani)
Persian/Farsi.
Urdu. etc...

**Comment by iiAmeer** [comment]
@jsvine im waiting....

**Comment by jsvine** [comment]
Thanks for your patience, @iiAmeer, and thank your July 16 response above. This is indeed something I'd like `pdfplumber` to handle well. It just happens to be a bit tricky.

**Comment by iiAmeer** [comment]
@jsvine it's actually easy (i think) , as the module loops on every character, just try to convert str to int. so when its an integer it won't be reversed. Pdfplumber  is my favourite pdf management tool So please do your best <3.

**Comment by moamen270** [comment]
I encountered the same issue with Arabic content, where some strings contain both text and numbers. While the numbers are displayed correctly, the text appears reversed. To address this, I created a function as a workaround.

```python
import re

def fix_arabic_with_numbers(text):
    """
    First reverses characters in each word, then reverses word order.
    Numbers remain unchanged.
    """
    if not isinstance(text, str):
        return text
    
    # Convert multiple spaces to single space and strip
    text = ' '.join(text.split())
    
    # Split into words
    words = text.split(' ')
    
    # Process each word - reverse characters but keep numbers
    fixed_words = []
    for word in words:
        # Split word into number and non-number parts
        parts = re.findall(r'\d+|[^\d]+', word)
        fixed_parts = []
        for part in parts:
            if part.isdigit():
                fixed_parts.append(part)  # Keep numbers as is
            else:
                fixed_parts.append(part[::-1])  # Reverse characters in text
        fixed_words.append(''.join(fixed_parts))
    
    # Reverse the order of words
    fixed_words = fixed_words[::-1]
    
    # Join words back together with spaces
    return ' '.join(fixed_words)
```

I hope that languages with RTL (Right-to-Left) will have support in the future.
